src/levanter/compat/torch_serialization.py

```diff
@@ -421,29 +420,15 @@ def get_to_cpu(arr):
                 r = np.array(arr)
                 return r
             else:
-                # unfortunately, jax's allgather seems to replicate to every device rather than every host
-                # which doesn't work for ~7B parameter models on TPU (assuming we also have optimizer state)
-                # this approach limits us to <64B parameters, but that's good enough for now
-                # we're going to do something a bit fancy, where we shard the model into a (process, device) mesh,
-                # then look for some axis along which we can shard the array, and then we'll do an allgather
-                # via pjit. If we can't find one, we'll just fully replicate since it probably isn't that big.
-                # TODO: ensure that this mesh arranges devices correctly
-                # (jax seems to do this internally itself, so we should be fine?)
-                process_mesh = Mesh(np.array(jax.devices()).reshape((jax.process_count(), -1)), ("process", "device"))
-                # now we need to find an axis along which we can shard the array.
-                # for this, we need to find an axis s.t. size(axis) % local_devices == 0
-
-                try:
-                    axis_to_shard = index_where(
-                        lambda axis_size: axis_size % process_mesh.devices.size == 0, arr.shape
-                    )
-                except ValueError:
-                    return np.array(arr)
-
-                shardings = [None if i != axis_to_shard else "device" for i in range(len(arr.shape))]
-                sharding = NamedSharding(process_mesh, PartitionSpec(*shardings))
-                out = jax.jit(_identity_fn, out_shardings=sharding)(arr)
-                return np.array(out)
+                shardings = [None for i in range(len(arr.shape))]
+                sharding = NamedSharding(arr.sharding.mesh, PartitionSpec(*shardings))
+
+                def _copy(in_array):
+                    return jax.lax.with_sharding_constraint(in_array, sharding)
+
+                # arr = jax.jit(_copy, donate_argnums=0)(arr)
+                arr = jax.device_put(arr, sharding)
```

Comment:
As per the deleted comment, I don't think this will work to serialize a 7B model (in fp32)?

Reply:
Ack, didn't mean to include this here, though I have an alternative solution which avoids some of the weird sharding issues. I'll put that in a separate PR to look at.